src/bun.js/api/server.zig

```diff
@@ -1235,6 +1279,7 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
         req: *uws.Request,
         signal: ?*JSC.WebCore.AbortSignal = null,
         method: HTTP.Method,
+        request_clones: ?ArrayList(*Request) = null,
```

Comment:
are you sure it's okay for this to be an `?ArrayList` and not a `?*ArrayList`? Is there any chance we might push to it and then have a stale reference? 

if so, can you leave a comment? i also don't think this needs to be a n`?` type, as that makes it go from 96 bytes to possibly 128 bytes. instead, we could check if .cap == 0

Reply:
how can you be confident of the lifetime of the `Request` pointer while it's inside this array? what prevents it from being garbage collected?

Reply:
@Jarred-Sumner https://github.com/oven-sh/bun/pull/6468/files#diff-e9484fda309708f80db81a125281dd0b9fdadaf3d2400ae8e78a2b8fc14d85ebR1336-R1345 and https://github.com/oven-sh/bun/pull/6468/files#diff-79f37ca42929ed1662aa69e017b6445a2cb38aa0a9e6c989a85e403eb32c07c9R287-R288